cmd/testwrapper/testwrapper.go

```diff
@@ -244,12 +249,21 @@ func main() {
 			fmt.Printf("\n\nAttempt #%d: Retrying flaky tests:\n\n", thisRun.attempt)
 		}
 
-		failed := false
 		toRetry := make(map[string][]string) // pkg -> tests to retry
 		for _, pt := range thisRun.tests {
 			ch := make(chan *testAttempt)
-			go runTests(ctx, thisRun.attempt, pt, otherArgs, ch)
+			runErr := make(chan error, 1)
+			go func() {
+				defer close(runErr)
+				runErr <- runTests(ctx, thisRun.attempt, pt, otherArgs, ch)
+			}()
+
+			var failed bool
 			for tr := range ch {
+				if tr.pkg == "command-line-arguments" {
```

Comment:
this should be documented somewhere higher up

Reply:
Documented in https://github.com/tailscale/tailscale/pull/9276/commits/b9d92b4ed64c1caa71883d9a264c9e5d17391276